File: classification/mstar_train.m
clear;
cd caffe
addpath(genpath('.\build\Release'));

% 设置gpu或cpu模式
% caffe.set_mode_cpu();
caffe.set_mode_gpu();
caffe.set_device(0);

% 设置solver
% solver = caffe.Solver('.\examples\mstar\mstar_solver.prototxt');
solver = caffe.Solver('.\examples\mstar\mstar_solver_96.prototxt');
% 训练
tic;
solver.solve();
toc;
rmpath(genpath('.\build\Release'));
cd ..

File: classification/mstar_train.py
import os
import sys
import time

root_dir = os.getcwd()
os.chdir('.\\caffe\\build\\Release')
sys.path.append('.\pycaffe')
import caffe

caffe.set_mode_gpu();
caffe.set_device(0);

os.chdir(root_dir + '.\caffe')
# solver = caffe.SGDSolver('.\examples\mstar\mstar_solver.prototxt');
solver = caffe.SGDSolver('.\examples\mstar\mstar_solver_96.prototxt');
start = time.clock()
solver.solve();
print('Time used: ' + str(round((time.clock() - start), 6)) + 's')